unix_integration/src/idprovider/interface.rs

```diff
@@ -63,11 +63,21 @@ pub struct UserToken {
 pub enum AuthCredHandler {
     Password,
     DeviceAuthorizationGrant,
+    MFA,
 }
 
 pub enum AuthRequest {
     Password,
-    DeviceAuthorizationGrant { data: DeviceAuthorizationResponse },
+    DeviceAuthorizationGrant {
+        data: DeviceAuthorizationResponse,
+    },
+    MFACode {
+        msg: String,
+        mfa_type: MFAType,
+        // Additional data required by the provider to complete the
+        // authentication, but not required by PAM
+        data: String,
```

Comment:
Maybe it should be Vec<u8> then to allow other encodings or structures? Why not just have more request variants for each type of possible additional data so we get typing?

Reply:
I like that idea.